src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -274,25 +264,38 @@ Napi::Value CoreWrap::get_available_devices(const Napi::CallbackInfo& info) {
 }
 
 Napi::Value CoreWrap::import_model(const Napi::CallbackInfo& info) {
-    if (info.Length() != 2) {
-        reportError(info.Env(), "Invalid number of arguments -> " + std::to_string(info.Length()));
-        return info.Env().Undefined();
-    }
-    if (!info[0].IsBuffer()) {
-        reportError(info.Env(), "The first argument must be of type Buffer.");
-        return info.Env().Undefined();
-    }
-    if (!info[1].IsString()) {
-        reportError(info.Env(), "The second argument must be of type String.");
+    try {
+        if (!info[0].IsBuffer()) {
+            OPENVINO_THROW("The first argument must be of type Buffer.");
+        }
+        if (!info[1].IsString()) {
+            OPENVINO_THROW("The second argument must be of type String.");
+        }
+        const auto& model_data = info[0].As<Napi::Buffer<uint8_t>>();
+        const auto model_stream = std::string(reinterpret_cast<char*>(model_data.Data()), model_data.Length());
+        std::stringstream _stream;
+        _stream << model_stream;
+
+        ov::CompiledModel compiled;
+        switch (info.Length()) {
+        case 2: {
+            compiled = _core.import_model(_stream, std::string(info[1].ToString()));
+            break;
+        }
+        case 3: {
+            compiled = _core.import_model(_stream, std::string(info[1].ToString()), to_anyMap(info.Env(), info[2]));
```

Comment:
Do you need to check `info[2]` that it's object first?

Reply:
I added this check to the conversion function. It is not required for choosing `import_model` overload